File: internal/dto/notification.go
package dto

type NotificationConfig struct {
	Enabled             bool     `json:"enabled"`
	WebhookURL          string   `json:"webhookUrl"`
	Secret              string   `json:"secret,omitempty"`
	PollIntervalMinutes int      `json:"pollIntervalMinutes,omitempty"`
	Scenarios           []string `json:"scenarios,omitempty"`
	ShopIDs             []string `json:"shopIds,omitempty"`
	AppID               string   `json:"appId,omitempty"`
	AppSecret           string   `json:"appSecret,omitempty"`
}

type NotificationState struct {
	LastRunAt        string            `json:"lastRunAt,omitempty"`
	LastRunOK        bool              `json:"lastRunOk"`
	LastError        string            `json:"lastError,omitempty"`
	LastSentCount    int               `json:"lastSentCount,omitempty"`
	LastBarcodeError string            `json:"lastBarcodeError,omitempty"`
	Notified         map[string]string `json:"notified,omitempty"`
}

type NotificationData struct {
	Config NotificationConfig `json:"config"`
	State  NotificationState  `json:"state"`
}

type NotificationConfigView struct {
	Enabled             bool     `json:"enabled"`
	WebhookURL          string   `json:"webhookUrl"`
	SecretSet           bool     `json:"secretSet"`
	PollIntervalMinutes int      `json:"pollIntervalMinutes"`
	Scenarios           []string `json:"scenarios"`
	ShopIDs             []string `json:"shopIds"`
	AppID               string   `json:"appId,omitempty"`
	AppSecretSet        bool     `json:"appSecretSet"`
}

type ScenarioOption struct {
	Key   string `json:"key"`
	Label string `json:"label"`
	Group string `json:"group,omitempty"`
}

type NotificationShopOption struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

type NotificationView struct {
	Config    NotificationConfigView   `json:"config"`
	State     NotificationState        `json:"state"`
	Scenarios []ScenarioOption         `json:"scenarios"`
	Shops     []NotificationShopOption `json:"shops"`
}

type NotificationRunResult struct {
	Sent             int    `json:"sent"`
	Skipped          int    `json:"skipped"`
	BarcodeWarnings  int    `json:"barcodeWarnings,omitempty"`
	LastBarcodeError string `json:"lastBarcodeError,omitempty"`
	Error            string `json:"error,omitempty"`
}

File: web/src/api/notification.ts
import client, { unwrap } from './client'

export interface NotificationScenarioOption {
  key: string
  label: string
  group?: string
}

export interface NotificationShopOption {
  id: string
  name: string
}

export interface NotificationConfig {
  enabled: boolean
  webhookUrl: string
  secret?: string
  secretSet?: boolean
  appId?: string
  appSecret?: string
  appSecretSet?: boolean
  pollIntervalMinutes: number
  scenarios: string[]
  shopIds?: string[]
}

export interface NotificationState {
  lastRunAt?: string
  lastRunOk?: boolean
  lastError?: string
  lastSentCount?: number
  lastBarcodeError?: string
}

export interface NotificationView {
  config: NotificationConfig
  state: NotificationState
  scenarios: NotificationScenarioOption[]
  shops: NotificationShopOption[]
}

export async function getNotification() {
  return unwrap<NotificationView>(await client.get('/notifications'))
}

export async function saveNotification(body: NotificationConfig) {
  return unwrap<NotificationView>(await client.put('/notifications', body))
}

export async function testNotification(text?: string) {
  return unwrap<{ ok: boolean }>(await client.post('/notifications/test', { text }))
}

export async function testBarcodeNotification() {
  return unwrap<{ ok: boolean }>(await client.post('/notifications/test-barcode'))
}

export async function runNotification() {
  return unwrap<{
    sent: number
    skipped: number
    barcodeWarnings?: number
    lastBarcodeError?: string
    error?: string
  }>(await client.post('/notifications/run'))
}

export async function resetNotificationState() {
  return unwrap<{ cleared: number; view: NotificationView }>(await client.post('/notifications/reset-state'))
}
